mosu-app/src/entities/profile/services/get-school-info.ts

```diff
@@ -0,0 +1,71 @@
+import axios from "axios";
+
+export interface SchoolInfoResponseBody {
+    SCHUL_NM: string; // 학교 이름 (schoolName)
+    ORG_RDNZC: string; // 학교 우편 번호 (zipcode)
+    ORG_RDNMA: string; // 학교 도로명 상세 주소 (street)
+}
+
+interface ApiResponse {
+    schoolInfo: Array<{
+        head: Array<{
+            list_total_count: number;
+            RESULT: {
+                CODE: string;
+                MESSAGE: string;
+            };
+        }>;
+        row: SchoolInfoResponseBody[];
+    }>;
+}
+
+export const getSchoolInfo = async (schoolName: string): Promise<SchoolInfoResponseBody[]> => {
+    if (!schoolName || schoolName.length < 2) {
+        return [];
+    }
+
+    const baseUrl = process.env.NEXT_PUBLIC_SCHOOL_INFO_URL;
+    const apiKey = process.env.NEXT_PUBLIC_OPEN_API_KEY;
+
+    if (!baseUrl || !apiKey) {
+        throw new Error("환경변수가 설정되지 않았습니다.");
+    }
+
+    try {
+        const { data } = await axios.get<ApiResponse>(baseUrl, {
+            params: {
+                KEY: apiKey,
+                Type: "json",
+                pIndex: 1,
+                pSize: 100,
+                SCHUL_NM: schoolName,
+                SCHUL_KND_SC_NM: "고등학교",
+            },
+        });
+
+        const headData = data.schoolInfo?.[0]?.head;
+
+        const resultCode = headData?.[1]?.RESULT?.CODE;
+
+        if (resultCode === "INFO-000") {
```

Comment:
head[1]에 접근하는 이유가 명확하지 않아 보여서 아래와 같이 수정하는 것이 더 명시적일 것 같아요! 어떻게 생각하시나요!

```suggestion
const result = data.schoolInfo?.[0]?.head?.[0]?.RESULT;
if (result?.CODE !== "INFO-000") return [];

```